tools/cpp/unix_cc_toolchain_config.bzl

```diff
@@ -891,7 +891,18 @@ def _impl(ctx):
                         iterate_over = "user_link_flags",
                         expand_if_available = "user_link_flags",
                     ),
-                ] + ([flag_group(flags = ctx.attr.link_libs)] if ctx.attr.link_libs else []),
+                ],
+            ),
+        ],
+    )
+
+    default_link_libs_feature = feature(
+        name = "default_link_libs",
+        enabled = True,
```

Comment:
Why is this enabled by default? The original one from which this was split isn't enabled by default.

Reply:
The starlark implementation of `user_link_flags` doesn't appear to be enabled by default here, yet if i try to disable it using `--features=-user_link_flags`, I get the error message:

```
The C++ toolchain '@local_config_cc//:toolchain' unconditionally implies feature 'user_link_flags', which is unsupported by this rule. This is most likely a misconfiguration in the C++ toolchain.
```

AFAICT, the feature is added unconditionally here: https://github.com/bazelbuild/bazel/blob/b21132597e90c42fc8dac10459b9326f2437ce5c/src/main/java/com/google/devtools/build/lib/rules/cpp/CppActionConfigs.java#L842

Reply:
That's adding it to the CROSSTOOL but not necessarily enabling it, I can't see where it is enabled by default.

Reply:
After inspecting the bazel codebase, it seems that the `user_link_flags` is marked as implied by the `c++-compile` action in the `getLegacyActionConfigs` function here:

https://github.com/bazelbuild/bazel/blob/cc55400e614e4e301244cc7c338ee3ea89424ce0/src/main/java/com/google/devtools/build/lib/rules/cpp/CppActionConfigs.java#L1113-L1129

These legacy actions are then added to the `actionConfigList` here:
https://github.com/bazelbuild/bazel/blob/cc55400e614e4e301244cc7c338ee3ea89424ce0/src/main/java/com/google/devtools/build/lib/rules/cpp/CcModule.java#L1224-L1235.

The `actionConfigList` is then provided to the CcToolchainConfigInfo constructor here:
https://github.com/bazelbuild/bazel/blob/cc55400e614e4e301244cc7c338ee3ea89424ce0/src/main/java/com/google/devtools/build/lib/rules/cpp/CcModule.java#L1245-L1262

A CcToolchainFeatures is created which uses this CcToolchainConfigInfo to assemble its `implies` member variable here:
https://github.com/bazelbuild/bazel/blob/cc55400e614e4e301244cc7c338ee3ea89424ce0/src/main/java/com/google/devtools/build/lib/rules/cpp/CcToolchainFeatures.java#L1612

The CcToolchainFeatures formsm part of the `FeatureConfiguration`, which is used in the following lines to detect unconditionally implied features:

https://github.com/bazelbuild/bazel/blob/cc55400e614e4e301244cc7c338ee3ea89424ce0/src/main/java/com/google/devtools/build/lib/rules/cpp/CcCommon.java#L1163-L1172



According to [this](https://docs.bazel.build/versions/main/cc-toolchain-config-reference.html#well-known-features) documentation, it should be possible to disable legacy features using the `no_legacy_features` feature, however this is not implemented in [unix_cc_configure.bzl ](https://github.com/bazelbuild/bazel/blob/master/tools/cpp/unix_cc_configure.bzl)